src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -75,17 +85,16 @@ private ResponseEntity response(final EventPublishResult result) {
         }
     }
 
-    private ResponseEntity doWithMetrics(final String eventTypeName, final long startingNanos,
-                                         final EventProcessingTask task) throws NakadiException {
+    private ResponseEntity doWithTimerMetric(final EventTypeMetrics eventTypeMetrics, final long startingNanos,
```

Comment:
The initial reason was that in the moment the timer is started I don't know if the process is going to succeed or fail. I want to report it differently, especially because we use the timer also as a counter. 

But after the refactoring it is not necessary to pass in the start time from outside. The `System.nanoTime()` can be called inside the method. I change it.